src/apify/storages/dataset.py

```diff
@@ -91,26 +146,17 @@ async def push_data(cls, data: JSONSerializable) -> None:
         return await dataset.push_data(data)
 
     async def _push_data_internal(self, data: JSONSerializable) -> None:
-        # const dispatch = (payload: string) => this.client.pushItems(payload);
-        # const limit = MAX_PAYLOAD_SIZE_BYTES - Math.ceil(MAX_PAYLOAD_SIZE_BYTES * SAFETY_BUFFER_PERCENT);
-
-        # // Handle singular Objects
-        # if (!Array.isArray(data)) {
-        #     const payload = checkAndSerialize(data, limit);
-        #     return dispatch(payload);
-        # }
+        # Handle singular items
+        if not isinstance(data, list):
+            payload = _check_and_serialize(data)
+            return await self._client.push_items(payload)
 
-        # // Handle Arrays
-        # const payloads = data.map((item, index) => checkAndSerialize(item, limit, index));
-        # const chunks = chunkBySize(payloads, limit);
+        # Handle lists
+        payloads = [_check_and_serialize(item, index) for index, item in enumerate(data)]
 
-        # // Invoke client in series to preserve order of data
-        # for (const chunk of chunks) {
-        #     await dispatch(chunk);
-        # }
-        # TODO: Implement the size chunking mechanism from crawlee...
-        # limit = MAX_PAYLOAD_SIZE_BYTES - math.ceil(MAX_PAYLOAD_SIZE_BYTES * SAFETY_BUFFER_PERCENT)
-        await self._client.push_items(data)
+        # Invoke client in series to preserve the order of data
+        for chunk in _chunk_by_size(payloads):
+            await self._client.push_items(chunk)
```

Comment:
This would mean that if a too large item would be in the middle of the data, then it would push some items and then fail in the middle, rather than failing before pushing anything. But I think that's an okay-enough tradeoff.

Reply:
Well another option is to check the whole list in one step and create a generator of serialized items in another step

Reply:
Then you'd have to serialize every item twice, that's not great either

Reply:
I changed the `payloads` list to a generator